### DISCUSSION TITLE
Several crystal phases from the same trained potential (polymorphism)

### DISCUSSION THREAD

**Discussion by mfhm** [deepmodeling/deepmd-kit#74] [Q&A]
Hi,

You have done a great job with deePMD code! Is it possible to train a potential, from which one could get different crystal phases depending on temperature and/or pressure? Are you aware of any publications, which have successfully demonstrated this?

Thanks in advance,
Max

**Comment by luigibonati** [comment]
Dear Max,

Yes, it is possible. You should take care of including not
only configurations from the bulk structures but also interfaces between
them as well as defects.
There are many approaches for this task. You can collect the configurations
just from plain MD, you can use active learning to select new
configurations based on ensemble uncertainty (see DeepMD-GEN) or you can
use enhanced sampling techniques to efficiently collect all the relevant
structures.
An example of the latter is the following:
https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.121.265701 . In
this case we have done it for the liquid and the cubic diamond structure of
silicon but this can be extended using the same procedure to larger portion
of phase diagrams. We have an another example which it's out on review, I
hope it will be out soon!

Best,
Luigi

**Comment by jameswind** [comment]
Thank you so much, Luigi, for your reply!
Hi Max, I confirm that the answer is yes and we have seen many successful
examples. Many of them are still under extensive tests, so my suggestion at
this moment is to take a look at Luigi's work on silicon and the active
learning approach (
https://journals.aps.org/prmaterials/abstract/10.1103/PhysRevMaterials.3.023804).
The latter realizes a PES model that is reliable for a large configuration
space, including several crystalline structures and liquids, for the Al-Mg
system. The approach is not limited to metallic systems. Please see the
Github repo dpgen (https://github.com/deepmodeling/dpgen) for more
information. dpgen is still under active development and any feedback or
contribution of yours would be greatly appreciated.

Best,
Linfeng
